---
title: "Auxiliary - Bioinformatics Project Meeting Agenda"
author: "FGCZ - (Draft)"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output: html_document
papersize: a4
geometry: margin=.5in
vignette: >
  %\VignetteIndexEntry{Auxiliary - Bioinformatics Project Meeting Agenda} 
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
  
---

\fontsize{10}{10}
\selectfont

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE, warning = FALSE, message = FALSE)
```


Project Nr :   

Order Id :     

Participants:

- User :
- Bioinformatician :
- Coach :


When?

- [ ]  before start of experiment
- [ ]  after QC sample preparation
- [ ]  after main experiment sample preparation

Comments : 


$~$

$~$

$~$

$~$


# Overview of the analysis


In label-free quantification (LFQ) experiments, we measure relative protein changes among groups of subjects. 
Before the main experiment, a quality control (QC) experiment is performed. We use the data from the QC to estimate the _within-group_ variance and to determine the number of samples in the main experiment. The measurements of the main experiment are modeled using linear models, mixed-effects linear models, which can be thought of as an extension of the t-test to more than two groups. These estimates are used to perform Gene Set Enrichment Analysis and Overrepresentation analysis using WebGestaltR and sigora. The meeting aims to obtain and document all the information relevant to perform a statistically sound LFQ data analysis. 



# Aim of the meeting

The goal of the meeting is this protocol specifying:

- which protein database to use
- which and how many samples to use for the QC
- determine all the parameters for the sample size estimation
- the design of your main experiment
- all the names of the factors and factor levels
- specify the hypothesis to be tested


The samples should be annotated with the names specified in the protocol in the b-fabric system. These names will also appear in the result tables and visualizations.

We perform the data analysis according to this protocol. Changing the analysis protocol (e.g. adding contrasts or blocking factors) after the analysis was delivered will incur time delays and additional costs.

# Genomics Analysis

Did you perform a genomics experiment on the same samples?

- [ ] no 
- [ ] yes


What type of experiment?

$~$

$~$

- Was it run at the FGCZ?
- project Id:
- order Id:
- Bioinformatician :
- Are you interested in Proteomics Genomics data integration?

$~$

$~$


Note:

- NGS data can be used to create improved protein identification databases. 
- If NGS - proteomics data integration is planned, compatible statistical analysis should be applied to the NGS and Proteomics data.




# Protein sequence database and downstream analysis tools

Which sequence database should be used for protein identification?

Which organism?

- [ ] homo sapiens
- [ ] mus musculus
- [ ] yeast
- [ ] zebra-fish
- other :


Should additional protein sequences be added to the database?

$~$

$~$


Contaminant list?

- [ ] Complete
- [ ] Without human contaminants
- other : 


Which downstream analysis tools will be used?

- [ ] [WebGestalt](http://www.webgestalt.org/){target="_blank"}  (WEB-based GEne SeT AnaLysis Toolkit)
- [ ] [string-db](https://string-db.org/){target="_blank"} (Known and predicted protein-protein interactions)
- other : 

$~$

Note:

- Typically we are using protein databases from uniprot.org

- Ideally the identifiers in the protein database are compatible with the downstream analysis tool you intend to use. For instance _WebGestalt_, although it recognizes many types of identifiers: e.g., Uniprot Swiss-Prot, Gene Symbols, NCBI Entrez gene, it does not work with Uniprot Treml identifiers.


# Effect size and size of test

These parameters are crucial for the QC experiment and will be assessed again when protein variance estimates from the QC experiment are available.
They also will be used when visualizing the results in the final report (e.g., Volcano plots) or when running overrepresentation analysis (ORA).


## Effect size

What is the smallest effect size (i.e., fold change in protein abundance) of biological relevance, you are interested in detecting? 
Typically fold changes are $1.5$, $2$, or $4$. This information is needed to determine the sample size as well as to biologically interpret the data.

The number of samples you need to use highly depends on the effect size you want to detect. Detecting small effect sizes requires you to measure more samples.
On the other hand, the statistical significance of a fold change does not imply biological relevance. Consequently, do not measure more samples than is required.


- [ ] 1.5
- [ ] 2 
- [ ] 4
- other :

## Size of the test[^5]

[^5]: In statistics, the size of a test is the probability of falsely rejecting the null hypothesis.

Typically a difference between two conditions is considered to be significant if the p-Value is less than $0.05$. For some applications a less significant p-Value can be used (e.g., $0.1$) for others a smaller p-Value might be required.

- [ ] 0.01
- [ ] 0.05
- [ ] 0.1
- other :

## Power of test

The power of a binary hypothesis test is the probability that the test rejects the null hypothesis (H0) when a specific alternative hypothesis (H1) is true [Wikipedia](https://en.wikipedia.org/wiki/Power_(statistics))

- [ ] 0.8
- other : 



## Does the variance within groups differ?

Is the variance within the groups the same or different? E.g., do you expect some groups with larger within group differences?

- [ ] yes
- [ ] no

If yes, which of the group's might have the highest within-group variance? (Some treatment might have heterogeneous outcomes.)

$~$
$~$




If possible, use samples from this group for the QC experiment, which will enable you to get a conservative estimate of the required samples sizes. If the variance in one group is substantially higher, the power of the tests can be improved by allocating more samples to this group.


## Difference in protein variances

From the QC experiment, we can estimate for each protein a  within-group variance. For some proteins, we observe more variation within a group compared to other proteins.

Are there proteins or protein groups of particular interest?

Provide protein IDs:  


Do you want to obtain significant p-Values for proteins with large within-group variances, or would you be satisfied with getting significances for the $X\%$ of low variance proteins? 


- [ ] $50\%$
- [ ] $70\%$
- other :


Note:

- High abundant proteins will exhibit smaller technical variability compared with low abundant proteins.

# Designs

This section covers the most frequently used designs.

- [ ] Parallel Group Design - Evaluates a single factor[^2]

- [ ] Factorial Design - Evaluates multiple factors simultaneously

## Parallel Group Design

[^2]: A parallel design, also called a parallel group study, compares two or more treatments. Participants are randomly assigned to either group, treatments are administered, and then the results are compared (statistics how to).

How many groups? 

- [ ] 2
- [ ] more. How many?   

Factor name (_camel case_[^1] e.g., Treatment) :  



Level names (_camel case_ e.g. Control, Diet, Creme): 


[^1]: _camel case_ - a typographical convention in which an initial capital is used for the first letter of a word-forming the second element of a closed compound, e.g., PayPal, iPhone, MasterCard.


## Factorial Design[^8]

Applies when studying how two or more factors influence protein expression, e.g., $Knockout$ and $Treatment$.


[^8]:In statistics, a full factorial experiment is an experiment whose design consists of two or more factors, each with discrete possible values or "levels", and whose experimental units take on all possible combinations of these levels across all such factors. Such an experiment allows the investigator to study the effect of each factor on the response variable, as well as the effects of interactions between factors on the response variable.

- How many factors? Typically 2 rarely 3.
- How many levels per factor? Typically 2 or 3 per factor.


| Factor Name  | Number levels  |   level names $~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~$|
|:-----|----|:----:|
|    $~$          |                |                     |
|     $~$         |                |                     |
|        $~$      |                |                     |
| $~$ | | |


- Provide computer-friendly names of the factors (grouping variables) in _camel case_.
- Provide names of levels for each grouping variable.

Example:

| Factor Name  | Number levels  |   level names $~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~$|
|:-----|----|:----:|
| Treatment |3 | Control, Diet, Creme| 
| AgeGroup | 2 | young, old |



## Repeated measurements


Applies when measurements are repeated on the same subject, e.g. over time [^3].

[^3]: Repeated measures design is a research design that involves multiple measures of the same variable taken on the same or matched subjects either under different conditions or over two or more time periods. For instance, repeated measurements are collected in a longitudinal study in which change over time is assessed (Wikipedia).

- [ ] Yes 
- [ ] No

If _yes_ please specify a computer-friendly factor name using _camel case_: e.g., PatientId.

Name of the factor : 

The subject identifiers can contain upper and lower case letters and numbers but avoid special characters and white spaces. 
Neither the factor names or the levels of the blocking factors will be shown in the results or figures generated.


## Blocking factors

A blocking factor is some variable that affects an experimental outcome but is itself of no interest, e.g., sample batch [^7]. Including blocking-factors can improve the statistical model. 



[^7]: A blocking factor is a factor used to create blocks. It is some variable that has an effect on an experimental outcome, but is itself of no interest. Blocking factors vary wildly depending on the experiment. For example: in human studies age or gender are often used as blocking factors. Blocking factors can be included into the model to describe the variance of the data [data science central](https://www.statisticshowto.datasciencecentral.com/blocking-factor/). 



| Factor Name  | Number levels  |
|:-----|----|
|    $~$          |                |
|     $~$         |                |
|        $~$      |                |
| $~$ | |



## Further information

$~$

$~$

$~$

$~$

$~$

$~$


# Contrasts - Hypothesis to be tested

If there are more than two groups, specify the hypothesis to be tested (comparison) using the factor and factor level names.


**_Contrasts to be computed:_** 

$~$

$~$

$~$

$~$

$~$

$~$

$~$

$~$

$~$

Examples for parallel Group design (only one factor):

- Diet - Control
- Creme - Control


Example for factorial designs:

- Treatment_Control - Treatment_Creme
- AgeGroup_young - AgeGroup_old
- Treatment_Control:AgeGroup_young - Treatment_Control:AgeGroup_old



Note:

- If one group is used in more comparisons than all other groups, it will benefit the power of the test [^6] to allocate more samples to this group, since increasing the number of degrees of freedom in this group will benefit many comparisons.

- __Limit the number of hypotheses you test__. Multiplicity - each comparison is a test of hypothesis. Therefore, it is required to perform a p-Value adjustment. For instance, using the Bonferroni correction, the obtained p-Values are multiplied by the number of the hypothesis tested. Given two comparisons, a p-Value of 0.03 after Bonferroni adjustment is $0.06$, i.e., not significant given a test size of $0.05$. Therefore, do not test all possible hypothesis! For instance, if including the contrast  _Diet - Creme_, it is required to multiply the p-Values with the factor $3$. i.e., a p-Value of $0.2$ after Bonferroni adjustment is $0.6$ - i.e., not significant.


[^6]:  The power of a binary hypothesis test is the probability that the test rejects the null hypothesis (H0) when a specific alternative hypothesis (H1) is true. The statistical power ranges from 0 to 1, and as statistical power increases, the probability of making a type II error (wrongly failing to reject the null hypothesis) decreases. [Wikipedia](https://en.wikipedia.org/wiki/Power_(statistics))

- Factorial designs generate many interactions (e.g., Treatment_Control:AgeGroup_young). A $3\times 2$ factorial design generates $6$ groups (unique combinations of factors) which makes it possible to specify many contrasts. Limit the number of hypotheses ideally to few (one or two) per experiment. It is required to adjust the p-Values based on the number of hypotheses examined.

- Typically in factorial designs, tests for differences among main effects (differences between levels within a single factor) and second-order interactions of the factors. 

- _main effects_ : Treatment_Control - Treatment_Diet 
- _Test for interaction_ : (Treatment_Control:AgeGroup_young - Treatment_Control:AgeGroup_old) - (Treatment_Diet:AgeGroup_young - Treatment_Diet:AgeGroup_old)



## More designs

- Time Series, e.g., growth curves - require at least 4-5 different well-designed time-points.
When less than 4-time points are measured than the time points are typically treated as factors and can be analyzed using our standard pipeline for factorial designs.

We might be able to support you with some other hypothesis tests, depending on our availability.

- Dose-response curves 
- Survival analysis 


# Other topics

Example: Normalization is applied to remove systematic differences in protein abundance due to different sample concentrations, or different amount of sample loaded on column. Normalization is important, so that true differentially expressed proteins can be detected.

$~$

$~$

$~$

$~$

$~$

$~$

$~$

$~$


Date: 

